### ISSUE TITLE
[rl] ppg categorical.rs: trunk recomputed twice per aux-phase minibatch

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#513]
- **Files:** `crates/rlevo-reinforcement-learning/src/algorithms/ppg/policies/categorical.rs`, `crates/rlevo-reinforcement-learning/src/algorithms/ppg/ppg_agent.rs`
- **Body:** Confirmed against current code. `PpgAuxValueHead::aux_value` and `PpgAuxValueHead::logits` (categorical.rs:107-117) each call the private `trunk()` helper independently, and each consumes `obs` by value. The live auxiliary-phase call site is `ppg_agent.rs:748-750`:

  ```rust
  let aux_v_pred = PpgAuxValueHead::aux_value(self.policy(), obs_t.clone()); // trunk #1
  let aux_v_loss = unclipped_value_loss(aux_v_pred, returns_t);
  let new_logits = PpgAuxValueHead::logits(self.policy(), obs_t);            // trunk #2
  let kl = policy_kl_categorical(old_logits_t, new_logits);
  ```

  Every auxiliary-phase minibatch runs the two-layer `tanh` trunk forward twice and registers two trunk autodiff subgraphs for `total.backward()`, when both heads branch off one trunk. Fix: add a joint `aux_forward(obs) -> (Tensor<B,1>, Tensor<B,2>)` (inherent on `PpgCategoricalPolicyHead<B>`, or a provided/overridable method on `PpgAuxValueHead`) that runs `trunk()` once and forwards both heads off the shared `h`.
- **Severity:** medium — real, confirmed double compute, but scoped to the PPG auxiliary phase only (not the per-env-step rollout path, which is the actual hottest path per #177).
- **Source:** Code review 2026-06-30
